### PR TITLE
downloadFile: downloading from another resource

### LINKED ISSUE #711: Using downloadFile to download a file of another resource
**Is your feature request related to a problem? Please describe.**

You can't download a file of another resource through downloadFile. Currently, the only way to do this is to export the downloadFile function in that resource.

**Describe the solution you'd like**

I suggest using the _:resourceName/path_ format, like fileOpen.

### PR DESCRIPTION
**Pull request by TheNormalnij**
Fix issue #711

Event [onClientFileDownloadComplete](https://wiki.multitheftauto.com/wiki/OnClientFileDownloadComplete) has parameter requestResource: resource that called downloadFile. The source of this event is the root element of the resource that download file.

### CODE REVIEW

**Comment by qaisjp** [comment]
@ccw808 — if I recall correctly, you once mentioned that you were unhappy with `downloadFile`. Do you think we should add new features to `downloadFile` in favour of encouraging other methods of sending files?

**Comment by TheNormalnij** [comment]
[Resources](https://drive.google.com/open?id=1m1v5IhwTfJ3iUt14qXsRx2Rev4yhcjTY) for testing.

<sub>_Note from admin: please try to upload files directly. Thanks!_ [resources.zip](https://github.com/multitheftauto/mtasa-blue/files/3193353/resources.zip)</sub>

**Comment by TheNormalnij** [comment]
I need a more flexible and convenient tool for downloading files. Now I have to use exports for this, but I don't see that this is a good way. Using `downloadFile` directly is more convenient.

**Comment by onikyannn** [comment]
Any updates?

**Comment by ghost** [comment]
Why dont you use fetchRemote ? I recently implemented abortRemoteRequest and getRemoteRequestInfo which both can handle downloads pretty well.

**Comment by TheNormalnij** [comment]
fetchRemote don't math CRC hash. Also, downloadFile use specific HTTP server from mtaserver.conf and don't use ACL right (function is safe).

**Comment by ccw808** [comment]
Looks good to me

**Review by ccw808 — APPROVED** [review]

File: Client/mods/deathmatch/logic/CSingularFileDownload.cpp [hunk 1]
```diff
@@ -65,6 +66,15 @@ void CSingularFileDownload::CallFinished(bool bSuccess)
...
         Arguments.PushString(GetShortName());            // file name
         Arguments.PushBoolean(bSuccess);                 // Completed successfully?
+        if (m_pRequestResource)
+        {
+            Arguments.PushResource(m_pRequestResource);  // Resource that called downloadFile
+        }
+        else
+        {
+            Arguments.PushBoolean(false);                 // or false
+        }
```
Comment (qaisjp) [inline Client/mods/deathmatch/logic/CSingularFileDownload.cpp:76]: @TheNormalnij, when is `m_pRequestResource` null?

Comment (TheNormalnij) [reply Client/mods/deathmatch/logic/CSingularFileDownload.cpp:76]: I think that m_pRequestResource can be deleted during downloading

**Pull request merged**